src/main/java/org/testng/DependencyMap.java

```diff
@@ -21,7 +22,20 @@
 
   public DependencyMap(ITestNGMethod[] methods) {
     for (ITestNGMethod m : methods) {
-    	m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
+      m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
```

Comment:
In fact, we are not supposed to do introspection in this class because it was already done before (current architecture of TestNG).
That means the problem is somewhere before.

Did you have a look on `MethodInheritance` or `MethodHelper` classes?